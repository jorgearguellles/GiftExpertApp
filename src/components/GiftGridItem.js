import PropTypes from 'prop-types';

const GiftGridItem = ({ title, urlImg }) => {

  return(
    <div className="card animate__animated animate__zoomIn">
      <img alt={title} src={urlImg} />
      <p>{title}</p>
    </div>
  )
};

export { GiftGridItem };

GiftGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  urlImg: PropTypes.string.isRequired,
}

GiftGridItem.defaultProps = {
  title: "Title here",
  urlImg: "https://www.google.com",
}

