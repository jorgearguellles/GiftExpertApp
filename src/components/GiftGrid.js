import { useFetchGifts } from "../hooks/useFetchGifts";
import { GiftGridItem } from "./GiftGridItem";
import PropTypes from 'prop-types';

const GiftGrid = ({ category }) => {

  const {data:images, loading} = useFetchGifts(category);

  return(
    <>
      <h4>{category}</h4>

      {loading && <p>Loading...</p>}
      
      <div className="card-grid">
        {
          images.map( img => ( <GiftGridItem key={img.id} {...img} />) )
        }
      </div> 
    </>
  )
};

export { GiftGrid };

GiftGrid.propTypes = {
  category: PropTypes.string.isRequired
}