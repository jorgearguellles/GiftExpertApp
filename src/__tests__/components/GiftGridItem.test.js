
import { GiftGridItem } from '../../components/GiftGridItem';
import { shallow } from "enzyme";

describe("< GiftGridItem /> testing",()=>{
  
  const title = "Title here";
  const url = "https://www.urlImages.com";
  const wrapper = shallow(<GiftGridItem title={title} urlImg={url} />);

  test("1. Should render correctly", ()=>{

    expect(wrapper).toMatchSnapshot();

  });

  test("2. Should have a P label with Title attribute", ()=>{

    const pLabel = wrapper.find('p');
    const textTitle = pLabel.text().trim()

    expect(textTitle).toBe(title);
  })

  test("3. Should to have a img-url = url by props & alt = alt by props", ()=>{

    const imgLabel = wrapper.find('img');

    // console.log(imgLabel.html()); // To see the element
    // console.log(imgLabel.props()); // to use the elements props

    // console.log(imgLabel.props().src); // A. to see SRC prop
    // console.log(imgLabel.prop('src')); // A'. to see SRC prop

    // console.log(imgLabel.props().alt); // B. to see ALT props
    // console.log(imgLabel.prop('alt')); // B'. to see ALT props

    const imgProp = imgLabel.prop('src');
    const altProp = imgLabel.prop('alt');

    expect(imgProp).toBe(url);
    expect(altProp).toBe(title);

  })

  test("4. Should to have just className = animate__zoomIn", ()=>{

    const divLabel = wrapper.find('div');
    const classN = divLabel.prop('className');
    
    expect(classN.includes('animate__zoomIn')).toBeTruthy();

  })

  test("5. Should to have a className = card animate__animated animate__zoomIn", ()=>{

    const divLabel = wrapper.find('div');
    const classN = 'card animate__animated animate__zoomIn'

    expect(divLabel.prop('className')).toBe(classN);

  })
})