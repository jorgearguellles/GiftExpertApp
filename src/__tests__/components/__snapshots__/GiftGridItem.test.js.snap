// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< GiftGridItem /> testing 1. Should render correctly 1`] = `
<div
  className="card animate__animated animate__zoomIn"
>
  <img
    alt="Title here"
    src="https://www.urlImages.com"
  />
  <p>
    Title here
  </p>
</div>
`;
